test/apps/sockaddr/sa_main.cc

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "sa_base.h"
+
+#include <iostream>
+#include <vector>
+#include <fstream>
+#include <cstring>
+#include <getopt.h>
+#include <netdb.h>
+
+
+class application {
+public:
+    class usage_exception : public error {
+    public:
+        usage_exception(const std::string& message = "");
+    };
+
+    application(int argc, char **argv);
+
+    int run();
+
+    static void usage(const std::string& error);
+
+private:
+    typedef std::pair<std::string, int> dest_t;
```

Comment:
hostname:port? pls add comment or better to declare a struct with corresponding fields

Reply:
ok